lib/auth/webauthnwin/api_test.go

```diff
@@ -0,0 +1,255 @@
+				cp := pwdlessOK
+				cp.Response.User.ID = nil
+				return &cp
+			},
+			wantErr: "user ID",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Millisecond)
+			defer cancel()
+
+			_, err := webauthnwin.Register(ctx, test.origin, test.createCC())
+			require.Error(t, err, "Register returned err = nil, want %q", test.wantErr)
+			assert.Contains(t, err.Error(), test.wantErr, "Register returned err = %q, want %q", err, test.wantErr)
+		})
+	}
+}
+
+func TestLogin_errors(t *testing.T) {
+	resetNativeAfterTests(t)
+
+	*webauthnwin.Native = fakeNative{}
+
+	const origin = "https://example.com"
+	okAssertion := &wanlib.CredentialAssertion{
+		Response: protocol.PublicKeyCredentialRequestOptions{
+			Challenge:      make([]byte, 32),
+			RelyingPartyID: "example.com",
+			AllowedCredentials: []protocol.CredentialDescriptor{
+				{Type: protocol.PublicKeyCredentialType, CredentialID: []byte{1, 2, 3, 4, 5}},
+			},
+		},
+	}
+
+	nilChallengeAssertion := *okAssertion
+	nilChallengeAssertion.Response.Challenge = nil
+
+	emptyRPIDAssertion := *okAssertion
+	emptyRPIDAssertion.Response.RelyingPartyID = ""
+
+	tests := []struct {
+		name      string
+		origin    string
+		assertion *wanlib.CredentialAssertion
+		wantErr   string
+	}{
+		{
+			// right now there is not need to provide anything in fakeNative
+			name:      "ok",
+			origin:    origin,
+			assertion: okAssertion,
+			wantErr:   "not implemented in fakeNative",
+		},
+		{
+			name:      "nil origin",
+			assertion: okAssertion,
+			wantErr:   "origin",
+		},
+		{
+			name:    "nil assertion",
+			origin:  origin,
+			wantErr: "assertion required",
+		},
+		{
+			name:      "assertion without challenge",
+			origin:    origin,
+			assertion: &nilChallengeAssertion,
+			wantErr:   "challenge",
+		},
+		{
+			name:      "assertion without RPID",
+			origin:    origin,
+			assertion: &emptyRPIDAssertion,
+			wantErr:   "relying party ID",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Millisecond)
+			defer cancel()
+
+			_, _, err := webauthnwin.Login(ctx, test.origin, test.assertion, nil /* opts */)
+			require.Error(t, err, "Login returned err = nil, want %q", test.wantErr)
+			assert.Contains(t, err.Error(), test.wantErr, "Login returned err = %q, want %q", err, test.wantErr)
+		})
+	}
+}
+
+func resetNativeAfterTests(t *testing.T) {
+	n := *webauthnwin.Native
+	t.Cleanup(func() {
+		*webauthnwin.Native = n
+	})
+}
+
+type fakeNative struct{}
+
+func (f fakeNative) CheckSupport() webauthnwin.CheckSupportResult {
+	return webauthnwin.CheckSupportResult{
+		HasCompileSupport: true,
+		IsAvailable:       true,
+	}
+}
+
+func (f fakeNative) GetAssertion(origin string, in *wanlib.CredentialAssertion, loginOpts *webauthnwin.LoginOpts) (*wanlib.CredentialAssertionResponse, error) {
```

Comment:
Let's write a happy-path test for login and registration. It's not much, but I'd rather have one.